### ISSUE TITLE
Mail: Filter and Forward Mail/Procmail Mail Filter

### ISSUE THREAD

**Issue by iliaross** [webmin/usermin#37]
Jamie, there is a bug for non English users.

How to reproduce:

  1. Go to _Usermin/Mail/Filter and Forward Mail_
  2. Click _Add New Mail Filter_
  3. Conditions: Based on header. Header let's say based on _Subject_, value that _Contains_, for example: _это просто тест_.
  4. Hit _Save_ at the bottom.

At first all looks fine but if you go to _Procmail Filter_ than you will see broken chars, which doesn't seem to work either, example:

![screenshot from 2018-04-04 14-13-19](https://user-images.githubusercontent.com/4426533/38304651-d08bf0ba-3812-11e8-9e2a-b66bb0afecca.png)

If I manually edit them in Procmail Mail Filter and hit save, then they are both displayed correctly in both modules.

Could you fix that?

**Comment by jcameron** [comment]
Seems like a unicode / character set issue. Which language and character set do you have selected in webmin?

**Comment by qooob** [comment]
Always _UTF-8_.

**Comment by qooob** [comment]
I'm talking about _Usermin_.

**Comment by jcameron** [comment]
Just tested this with Usermin in `en.UTF-8` language, but I was able to create a filter with the same text you used just fine.

One warning though - in practice this probably wouldn't work anyway, as email headers can only be in ASCII without some special encoding (that procmail won't match on).

**Comment by qooob** [comment]
No, Jamie, you misunderstood. It's displayed correctly in _Filter and Forward Mail_ but in _Procmail Mail Filter_ it's not displayed correctly after adding a rule in _Filter and Forward Mail_.

To reproduce it add it in _Filter and Forward Mail_, then go to _Procmail Mail Filter_ and you will see abracadabra.

```
Match regexp ^From: \�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\ \�\�\�\�\�\�\�\�\�\�\�\�\ \�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�\�
```

In case it's still true and Procmail can't do it, we need to inform a user with an error, explaining it, not just failing silently.

I will test it more and share results. It's hard to believe that it wouldn't work with unicode headers in 2018.

**Comment by jcameron** [comment]
Ok, I see the cause of this now, and will fix it in the next release. The real issue is that the string written to .procmailrc has a bunch of extra un-necessary backslashes in it.

I'm still not sure if putting unicode in .procmailrc will work though. You should try it out on some real emails by editing .procmailrc directly.

For example, here's the Subject: header from a chinese spam message I received recently : 

```
Subject: [SPAM] =?utf-8?B?NOaciOWfueiureebruaghyDmlrDku7vnu4/nkIbjgIHpg6jpl6jkuLvnrqHnrqHnkIbmioDog70=?=
	=?utf-8?B?5o+Q5Y2H6K6t57uD?=
```

**Comment by qooob** [comment]
Okay, correct. It doesn't work with non-Lating alphabet.

Jamie, could you test the input data for header and in case it contains non-Latin chars, could you automatically add conversion, so

```
это просто тест
```

would become:

```
=?UTF-8?B?0Y3RgtC+INC/0YDQvtGB0YLQviDRgtC10YHRgg==?=
```

That isn't hard to do?

**Comment by jcameron** [comment]
That should be possible, although I can see it getting complex if the user enters a regexp that involves unicode characters.

**Comment by qooob** [comment]
Maybe just starting from the simple part would be enough, usually having just a string is fine. In case you can't make it for regexp you could return the message, something like `Regexp not supported for Unicode characters`. It could be whatever as long as it doesn't fail silently.

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
